azurerm/internal/services/digitaltwins/parse/digital_twins_endpoint.go

```diff
@@ -0,0 +1,55 @@
+package parse
+
+// NOTE: this file is generated via 'go:generate' - manual changes will be overwritten
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type DigitalTwinsEndpointId struct {
+	SubscriptionId           string
+	ResourceGroup            string
+	DigitalTwinsInstanceName string
+	EndpointName             string
+}
+
+func NewDigitalTwinsEndpointID(subscriptionId, resourceGroup, digitalTwinsInstanceName, endpointName string) DigitalTwinsEndpointId {
+	return DigitalTwinsEndpointId{
+		SubscriptionId:           subscriptionId,
+		ResourceGroup:            resourceGroup,
+		DigitalTwinsInstanceName: digitalTwinsInstanceName,
+		EndpointName:             endpointName,
+	}
+}
+
+func (id DigitalTwinsEndpointId) ID(_ string) string {
+	fmtString := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.DigitalTwins/digitalTwinsInstances/%s/endpoints/%s"
+	return fmt.Sprintf(fmtString, id.SubscriptionId, id.ResourceGroup, id.DigitalTwinsInstanceName, id.EndpointName)
+}
+
+func DigitalTwinsEndpointID(input string) (*DigitalTwinsEndpointId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceId := DigitalTwinsEndpointId{
+		SubscriptionId: id.SubscriptionID,
+		ResourceGroup:  id.ResourceGroup,
+	}
+
+	if resourceId.DigitalTwinsInstanceName, err = id.PopSegment("digitalTwinsInstances"); err != nil {
```

Comment:
same above

Reply:
Since it is auto generated we should prolly leave it alone for now. I will speak to Tom about how to go about maybe exposing a naming override since in some cases the name may not make sense or be unnecessarily long.